app/models/user.rb

```diff
@@ -1,8 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
   validates :email, presence: true, uniqueness: true,
             format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i }
-  validates :password, presence: true, length: { in: 8..20,
-                                                 too_long: 'is too long.Maximum is 20',
-                                                 too_short: 'is too short. Minimum is 8' },
-            unless: Proc.new { |a| a.password.blank? }
+  validates :encrypted_password, :first_name, :last_name, presence: true
```

Comment:
We don't need to validate `encrypted_password` and also email as both are handled by devise, you can test it out.